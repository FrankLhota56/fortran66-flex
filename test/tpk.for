C     THE TPK ALGORITHM
C     FORTRAN IV STYLE
      DIMENSION A(11)
      FUN(T) = SQRT(ABS(T)) + 5.0*T**3
      READ (5, 1) A
  1   FORMAT(5F10.2)
      DO 10 J = 1, 11
         I = 11 - J
         Y = FUN(A(I+1))
         IF (400.0-Y) 4, 8, 8
  4         WRITE (6,5) I
  5         FORMAT(I10, 10H TOO LARGE)
         GO TO 10
  8         WRITE (6,9) I, Y
  9         FORMAT(I10, F12.6)
 10   CONTINUE
      STOP
      END
