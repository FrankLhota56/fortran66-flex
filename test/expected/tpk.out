Line     3: DIMENSION
Line     3: ID(a)
Line     3: '('
Line     3: INT_LIT(11)
Line     3: ')'
Line     4: EOS
Line     4: ID(fun)
Line     4: '('
Line     4: ID(t)
Line     4: ')'
Line     4: '='
Line     4: ID(sqrt)
Line     4: '('
Line     4: ID(abs)
Line     4: '('
Line     4: ID(t)
Line     4: ')'
Line     4: ')'
Line     4: '+'
Line     4: REAL_LIT(5.000000)
Line     4: '*'
Line     4: ID(t)
Line     4: EXPONENTIATION
Line     4: INT_LIT(3)
Line     5: EOS
Line     5: READ
Line     5: '('
Line     5: INT_LIT(5)
Line     5: ','
Line     5: INT_LIT(1)
Line     5: ')'
Line     5: ID(a)
Line     6: EOS
Line     6: INT_LIT(1)
Line     6: FORMAT
Line     6: '('
Line     6: FIELD_DESC(0P5F10.2)
Line     6: ')'
Line     7: EOS
Line     7: DO
Line     7: INT_LIT(10)
Line     7: ID(j)
Line     7: '='
Line     7: INT_LIT(1)
Line     7: ','
Line     7: INT_LIT(11)
Line     8: EOS
Line     8: ID(i)
Line     8: '='
Line     8: INT_LIT(11)
Line     8: '-'
Line     8: ID(j)
Line     9: EOS
Line     9: ID(y)
Line     9: '='
Line     9: ID(fun)
Line     9: '('
Line     9: ID(a)
Line     9: '('
Line     9: ID(i)
Line     9: '+'
Line     9: INT_LIT(1)
Line     9: ')'
Line     9: ')'
Line    10: EOS
Line    10: IF
Line    10: '('
Line    10: REAL_LIT(400.000000)
Line    10: '-'
Line    10: ID(y)
Line    10: ')'
Line    10: INT_LIT(4)
Line    10: ','
Line    10: INT_LIT(8)
Line    10: ','
Line    10: INT_LIT(8)
Line    11: EOS
Line    11: INT_LIT(4)
Line    11: WRITE
Line    11: '('
Line    11: INT_LIT(6)
Line    11: ','
Line    11: INT_LIT(5)
Line    11: ')'
Line    11: ID(i)
Line    12: EOS
Line    12: INT_LIT(5)
Line    12: FORMAT
Line    12: '('
Line    12: FIELD_DESC(1I10)
Line    12: ','
Line    12: HOLLERITH_LIT( TOO LARGE)
Line    12: ')'
Line    13: EOS
Line    13: GO
Line    13: TO
Line    13: INT_LIT(10)
Line    14: EOS
Line    14: INT_LIT(8)
Line    14: WRITE
Line    14: '('
Line    14: INT_LIT(6)
Line    14: ','
Line    14: INT_LIT(9)
Line    14: ')'
Line    14: ID(i)
Line    14: ','
Line    14: ID(y)
Line    15: EOS
Line    15: INT_LIT(9)
Line    15: FORMAT
Line    15: '('
Line    15: FIELD_DESC(1I10)
Line    15: ','
Line    15: FIELD_DESC(0P1F12.6)
Line    15: ')'
Line    16: EOS
Line    16: INT_LIT(10)
Line    16: CONTINUE
Line    17: EOS
Line    17: STOP
Line    18: EOS
Line    18: END
